Please allow use of custom simde location

VCV Rack decided to go with simde for SSE compat under ARM, and Cardinal is following that (while initially using sse2neon instead).
Due to differences on the version shipped with Rack vs the one shipped with surge, there can be symbol conflicts.
Doing an LTO shows this message:

```
../include/simd-compat/../simde/simde/x86/sse2.h:136:3: error: type 'union simde__m128i_private' violates the C++ One Definition Rule [-Werror=odr]
plugins/surgext/surge/src/../libs/simde/simde/x86/sse2.h:124:3: note: a different type is defined in another translation unit
  124 | } simde__m128i_private;
      |   ^
../include/simd-compat/../simde/simde/x86/sse2.h:95:38: note: the first difference of corresponding definitions is field 'neon_f16'
plugins/surgext/surge/src/../libs/simde/simde/x86/sse2.h:94:38: note: a field with different name is defined in another translation unit
   94 |     SIMDE_ALIGN_TO_16 float32x4_t    neon_f32;
```

A possible solution is to be able to use a custom simde version, alike done with juce.
That should be the approach for surge-rack too IMO, since the library is already provided by the Rack SDK modules built against VCV Rack should use it instead of their own custom version.

Alternatively the built-in simde could be contained in a custom namespace, but not sure how much work that is..

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -38,8 +38,9 @@ else()
   set(SURGE_USE_ALSA TRUE)
 endif()
 
+set(SURGE_SIMDE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../libs/simde" CACHE STRING "Path to simde library source tree")
 add_library(simde INTERFACE)
-target_include_directories(simde INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/../libs/simde)
+target_include_directories(simde INTERFACE ${SURGE_SIMDE_PATH})
 add_library(surge::simde ALIAS simde)
 
 add_library(surge-juce INTERFACE)

